src/java.base/share/classes/java/net/SocketPermission.java

```diff
@@ -651,28 +652,23 @@ private boolean isUntrusted()
      * attempt to get the fully qualified domain name
      *
      */
-    void getCanonName()
-        throws UnknownHostException
-    {
-        if (cname != null || invalid || untrusted) return;
-
+     void getCanonName() throws UnknownHostException {
+         if (cnames != null || invalid || untrusted) {
+            return;
+         }
         // attempt to get the canonical name
-
         try {
-            // first get the IP addresses if we don't have them yet
-            // this is because we need the IP address to then get
-            // FQDN.
-            if (addresses == null) {
-                getIP();
-            }
-
```

Comment:
The block above 665:667 is still needed. getIP() must be called to set the addresses field. There's a test that checks this.